ui_testing/pages/base_pages.py

```diff
@@ -326,3 +327,45 @@ def cancel_overview_pop_up(self):
         self.base_selenium.click(element='general:cancel_overview')
         self.sleep_tiny()
 
+    def open_pagination_menu(self):
+        self.base_selenium.wait_element(element='general:pagination_button')
+        self.base_selenium.click(element='general:pagination_button')
+
+    def set_page_limit(self, limit='20'):
+        self.base_selenium.LOGGER.info('set the pagination limit to {}'.format(limit))
+        self.open_pagination_menu()
+        limit_index = self.pagination_elements_array.index(limit)
+        self.base_selenium.wait_element(element='general:pagination_menu')
+        pagination_elements = self.base_selenium.find_elements_in_element(source_element='general:pagination_menu', destination_element='general:li')
+        if limit_index >= 0:
+            pagination_elements[limit_index].click()
+        
+
+    def get_current_pagination_limit(self):
+        return self.base_selenium.find_element(element='general:pagination_button').text.split('\n')[0]
+
+    # eslam, i'll need your check on this function
+    def wait_for_loading_msg(self):
+        self.base_selenium.LOGGER.info('wait for loading msg to disappear')
+        self.base_selenium.wait_element('general:loading_msg')
+        self.base_selenium.wait_until_element_is_not_displayed('general:loading_msg')
```

Comment:
@0xIslamTaha this I didn't create this method it didn't belong to me

Reply:
islam, i am trying to wait for loading symbol to appear and then wait for it to disappear, to make sure that data are loaded successfully, so the waiting is dynamically changed with the data loading instead of using sleep as we agreed in the meeting

Reply:
okay